src/eduid_webapp/group_management/views/invite.py

```diff
@@ -0,0 +1,209 @@
+# POSSIBILITY OF SUCH DAMAGE.
+#
+from typing import Mapping
+from uuid import UUID
+
+from flask import Blueprint
+from pymongo.errors import DuplicateKeyError
+
+from eduid_common.api.decorators import MarshalWith, UnmarshalWith, require_user
+from eduid_common.api.exceptions import MailTaskFailed
+from eduid_common.api.messages import CommonMsg, error_message
+from eduid_groupdb import User as GraphUser
+from eduid_scimapi.userdb import ScimApiUser
+from eduid_userdb import User
+from eduid_userdb.exceptions import DocumentDoesNotExist, EduIDDBError
+from eduid_userdb.group_management import GroupInviteState
+
+from eduid_webapp.group_management.app import current_group_management_app as current_app
+from eduid_webapp.group_management.helpers import (
+    GroupManagementMsg,
+    accept_group_invitation,
+    get_incoming_invites,
+    get_outgoing_invites,
+    get_scim_user_by_eppn,
+    is_owner,
+    send_invite_email,
+)
+from eduid_webapp.group_management.schemas import (
+    GroupAllInviteResponseSchema,
+    GroupIncomingInviteResponseSchema,
+    GroupInviteRequestSchema,
+    GroupOutgoingInviteResponseSchema,
+)
+
+__author__ = 'lundberg'
+
+group_invite_views = Blueprint('group_invite', __name__, url_prefix='/invites/', template_folder='templates')
+
+
+@group_invite_views.route('/all', methods=['GET'])
+@MarshalWith(GroupAllInviteResponseSchema)
+@require_user
+def all_invites(user: User) -> Mapping:
+    incoming = get_incoming_invites(user)
+    outgoing = []
+    scim_user = get_scim_user_by_eppn(user.eppn)
+    if scim_user:
+        # The user can only be a group owner if there is a scim_user
+        graph_user = GraphUser(identifier=str(scim_user.scim_id))
+        owner_groups = current_app.scimapi_groupdb.get_groups_for_owner(owner=graph_user)
+        outgoing = get_outgoing_invites(owner_groups)
+
+    return {'incoming': incoming, 'outgoing': outgoing}
+
+
+@group_invite_views.route('/incoming', methods=['GET'])
+@MarshalWith(GroupIncomingInviteResponseSchema)
+@require_user
+def incoming_invites(user: User) -> Mapping:
+    invites = get_incoming_invites(user)
+    return {'incoming': invites}
+
+
+@group_invite_views.route('/outgoing', methods=['GET'])
+@MarshalWith(GroupOutgoingInviteResponseSchema)
+@require_user
+def outgoing_invites(user: User) -> Mapping:
+    invites = []
+    scim_user = get_scim_user_by_eppn(user.eppn)
+    if scim_user:
+        # The user can only be a group owner if there is a scim_user
+        graph_user = GraphUser(identifier=str(scim_user.scim_id))
+        owner_groups = current_app.scimapi_groupdb.get_groups_for_owner(owner=graph_user)
+        invites = get_outgoing_invites(owner_groups)
+
+    return {'outgoing': invites}
+
+
+@group_invite_views.route('/create', methods=['POST'])
+@UnmarshalWith(GroupInviteRequestSchema)
+@MarshalWith(GroupOutgoingInviteResponseSchema)
+@require_user
+def create_invite(user: User, identifier: UUID, email_address: str, role: str) -> Mapping:
```

Comment:
maybe 'identifier' would be better named group_scim_id, or group_identifier, or group_id?